android/src/com/mapswithme/maps/MwmActivity.java

```diff
@@ -1426,9 +1453,12 @@ private void showMainMenu(boolean show)
   @Override
   public void onRoutingPlanStartAnimate(boolean show)
   {
-    int totalHeight = calcFloatingViewsOffset();
-    adjustCompassAndTraffic(!show ? UiUtils.getStatusBarHeight(getApplicationContext())
-                                  : totalHeight);
+    int offset;
+    if (!show || mRoutingPlanInplaceController == null
+        || (offset = mRoutingPlanInplaceController.calcHeight()) == 0)
```

Comment:
It looks strange to assign offset here and then reassign it immediately in the next line.

Reply:
If zero offset is not good (like in previous calcFloatingViewsOffset() function), this logic with always call adjustCompassAndTraffic looks suspicious

Reply:
I'll rework this logic for something simpler. I simply copy-pasted the old logic without trying to refactor it in fear of breaking something.